### ISSUE TITLE
Flutter Gallery Project can't be build any longer

### ISSUE THREAD

**Issue by vivek0103** [flutter/flutter#22467]
cupertino_picker_demo.dart results in bulid Error.

https://github.com/flutter/flutter/commit/2a8e35cc9c42e6957b069439eaa56207afed5ea2

The latest fix checked in breaks the build with following error.

**Compiler message:**
```
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:168:23: Error: Getter not found: 'CupertinoDatePickerMode'.
                mode: CupertinoDatePickerMode.date,
                      ^^^^^^^^^^^^^^^^^^^^^^^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:167:15: Error: Method not found: 'CupertinoDatePicker'.
              CupertinoDatePicker(
              ^^^^^^^^^^^^^^^^^^^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:167:15: Error: The method 'CupertinoDatePicker' isn't defined for the class '#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing method, or defining a method named 'CupertinoDatePicker'.
              CupertinoDatePicker(
              ^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:168:23: Error: The getter 'CupertinoDatePickerMode' isn't defined for the class '#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoDatePickerMode'.
                mode: CupertinoDatePickerMode.date,
                      ^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:198:23: Error: Getter not found: 'CupertinoDatePickerMode'.
                mode: CupertinoDatePickerMode.time,
                      ^^^^^^^^^^^^^^^^^^^^^^^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:197:15: Error: Method not found: 'CupertinoDatePicker'.
              CupertinoDatePicker(
              ^^^^^^^^^^^^^^^^^^^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:197:15: Error: The method 'CupertinoDatePicker' isn't defined for the class '#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing method, or defining a method named 'CupertinoDatePicker'.
              CupertinoDatePicker(
              ^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:198:23: Error: The getter 'CupertinoDatePickerMode' isn't defined for the class '#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoDatePickerMode'.
                mode: CupertinoDatePickerMode.time,
                      ^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:228:23: Error: Getter not found: 'CupertinoDatePickerMode'.
                mode: CupertinoDatePickerMode.dateAndTime,
                      ^^^^^^^^^^^^^^^^^^^^^^^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:227:15: Error: Method not found: 'CupertinoDatePicker'.
              CupertinoDatePicker(
              ^^^^^^^^^^^^^^^^^^^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:227:15: Error: The method 'CupertinoDatePicker' isn't defined for the class '#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing method, or defining a method named 'CupertinoDatePicker'.
              CupertinoDatePicker(
              ^
file:///d:/Repos/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart:228:23: Error: The getter 'CupertinoDatePickerMode' isn't defined for the class '#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoDatePickerMode'.
                mode: CupertinoDatePickerMode.dateAndTime,
                      ^
Compiler failed on d:\Repos\flutter\examples\flutter_gallery\lib/main.dart
Gradle task 'assembleDebug'... Done                         10.1s
Gradle task assembleDebug failed with exit code 1
```

**Comment by xster** [comment]
I'm assuming you're on the master channel and manually running git fetch/pull to update your Flutter repo? Try `flutter clean` and `flutter run` again?

**Comment by zoechi** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!
Thanks for your contribution.

**Comment by alexleeelkins** [comment]
@zoechi having issues with this myself, now. what information do you need?

`~/D/A/flutter.nosync    examples/flutter_gallery  flutter run                                       849ms  085249 Thu 181220
Launching lib/main.dart on iPhone Xʀ in debug mode...

Compiler message:
lib/gallery/app.dart:145:21: Error: Method not found: 'CupertinoThemeData'.
              data: CupertinoThemeData(
                    ^^^^^^^^^^^^^^^^^^
lib/gallery/app.dart:144:13: Error: Method not found: 'CupertinoTheme'.
            CupertinoTheme(
            ^^^^^^^^^^^^^^
lib/gallery/app.dart:144:13: Error: The method 'CupertinoTheme' isn't defined for the class '#lib1::_GalleryAppState'.
Try correcting the name to the name of an existing method, or defining a method named 'CupertinoTheme'.
            CupertinoTheme(
            ^^^^^^^^^^^^^^
lib/gallery/app.dart:145:21: Error: The method 'CupertinoThemeData' isn't defined for the class '#lib1::_GalleryAppState'.
Try correcting the name to the name of an existing method, or defining a method named 'CupertinoThemeData'.
              data: CupertinoThemeData(
                    ^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:53:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:62:35: Error: Method not found: 'CupertinoButton.filled'.
                  CupertinoButton.filled(
                                  ^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:90:35: Error: Method not found: 'CupertinoButton.filled'.
                  CupertinoButton.filled(
                                  ^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:119:35: Error: Method not found: 'CupertinoButton.filled'.
                  CupertinoButton.filled(
                                  ^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:135:35: Error: Method not found: 'CupertinoButton.filled'.
                  CupertinoButton.filled(
                                  ^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:146:35: Error: Method not found: 'CupertinoButton.filled'.
                  CupertinoButton.filled(
                                  ^^^^^^
lib/demo/cupertino/cupertino_alert_demo.dart:53:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoAlertDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_buttons_demo.dart:31:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_buttons_demo.dart:69:37: Error: Method not found: 'CupertinoButton.filled'.
                    CupertinoButton.filled(
                                    ^^^^^^
lib/demo/cupertino/cupertino_buttons_demo.dart:76:43: Error: Method not found: 'CupertinoButton.filled'.
                    const CupertinoButton.filled(
                                          ^^^^^^
lib/demo/cupertino/cupertino_buttons_demo.dart:31:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoButtonDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:55:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:55:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::CupertinoNavigationDemo'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:349:45: Error: Method not found: 'CupertinoButton.filled'.
                            CupertinoButton.filled(
                                            ^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:363:45: Error: Method not found: 'CupertinoButton.filled'.
                            CupertinoButton.filled(
                                            ^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:718:18: Error: Getter not found: 'CupertinoTheme'.
          color: CupertinoTheme.of(context).brightness == Brightness.light
                 ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:719:33: Error: Getter not found: 'extraLightBackgroundGray'.
              ? CupertinoColors.extraLightBackgroundGray
                                ^^^^^^^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:720:33: Error: Getter not found: 'darkBackgroundGray'.
              : CupertinoColors.darkBackgroundGray,
                                ^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:736:26: Error: Getter not found: 'CupertinoTheme'.
                  color: CupertinoTheme.of(context).scaffoldBackgroundColor,
                         ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:752:51: Error: Getter not found: 'CupertinoTheme'.
                          style: TextStyle(color: CupertinoTheme.of(context).primaryColor),
                                                  ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:718:18: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::CupertinoDemoTab3'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
          color: CupertinoTheme.of(context).brightness == Brightness.light
                 ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:736:26: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::CupertinoDemoTab3'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
                  color: CupertinoTheme.of(context).scaffoldBackgroundColor,
                         ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:752:51: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::CupertinoDemoTab3'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
                          style: TextStyle(color: CupertinoTheme.of(context).primaryColor),
                                                  ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_navigation_demo.dart:790:29: Error: Method not found: 'CupertinoButton.filled'.
            CupertinoButton.filled(
                            ^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:37:16: Error: Getter not found: 'CupertinoTheme'.
        color: CupertinoTheme.of(context).scaffoldBackgroundColor,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:37:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        color: CupertinoTheme.of(context).scaffoldBackgroundColor,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:254:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:257:20: Error: Getter not found: 'CupertinoTheme'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                   ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:257:61: Error: Getter not found: 'Brightness'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                                                            ^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:258:35: Error: Getter not found: 'extraLightBackgroundGray'.
                ? CupertinoColors.extraLightBackgroundGray
                                  ^^^^^^^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:259:35: Error: Getter not found: 'darkBackgroundGray'.
                : CupertinoColors.darkBackgroundGray,
                                  ^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:254:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:257:20: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                   ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_picker_demo.dart:257:61: Error: The getter 'Brightness' isn't defined for the class
'#lib1::_CupertinoPickerDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'Brightness'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                                                            ^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:42:14: Error: Getter not found: 'CupertinoTheme'.
      style: CupertinoTheme.of(context).textTheme.textStyle,
             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:46:20: Error: Getter not found: 'CupertinoTheme'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                   ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:46:61: Error: Getter not found: 'Brightness'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                                                            ^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:47:35: Error: Getter not found: 'extraLightBackgroundGray'.
                ? CupertinoColors.extraLightBackgroundGray
                                  ^^^^^^^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:48:35: Error: Getter not found: 'darkBackgroundGray'.
                : CupertinoColors.darkBackgroundGray,
                                  ^^^^^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:42:14: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoRefreshControlDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
      style: CupertinoTheme.of(context).textTheme.textStyle,
             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:46:20: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoRefreshControlDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                   ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:46:61: Error: The getter 'Brightness' isn't defined for the class
'#lib1::_CupertinoRefreshControlDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'Brightness'.
            color: CupertinoTheme.of(context).brightness == Brightness.light
                                                            ^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:157:14: Error: Getter not found: 'CupertinoTheme'.
      color: CupertinoTheme.of(context).scaffoldBackgroundColor,
             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:224:30: Error: Getter not found: 'CupertinoTheme'.
                      color: CupertinoTheme.of(context).primaryColor,
                             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:157:14: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_ListItem'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
      color: CupertinoTheme.of(context).scaffoldBackgroundColor,
             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_refresh_demo.dart:224:30: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_ListItem'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
                      color: CupertinoTheme.of(context).primaryColor,
                             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_segmented_control_demo.dart:63:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_segmented_control_demo.dart:94:30: Error: Getter not found: 'CupertinoTheme'.
                      color: CupertinoTheme.of(context).scaffoldBackgroundColor,
                             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_segmented_control_demo.dart:63:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoSegmentedControlDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_segmented_control_demo.dart:94:30: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoSegmentedControlDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
                      color: CupertinoTheme.of(context).scaffoldBackgroundColor,
                             ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_slider_demo.dart:32:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_slider_demo.dart:32:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoSliderDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_switch_demo.dart:32:16: Error: Getter not found: 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
lib/demo/cupertino/cupertino_switch_demo.dart:32:16: Error: The getter 'CupertinoTheme' isn't defined for the class
'#lib1::_CupertinoSwitchDemoState'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'CupertinoTheme'.
        style: CupertinoTheme.of(context).textTheme.textStyle,
               ^^^^^^^^^^^^^^
Compiler failed on /Users/alex/Documents/AlexElkins.xyz/flutter.nosync/examples/flutter_gallery/lib/main.dart
Error launching application on iPhone Xʀ.`

**Comment by zoechi** [comment]
Looks like you are using `master` channel where Gallery code is using Flutter code that is not merged yet.
@xster Seems you committed CupertionTheme yesterday. You might know more.

**Comment by zoechi** [comment]
> what information do you need?

at least the output of `flutter doctor -v`

**Comment by jinfagang** [comment]
@zoechi  OK, I just pulled the fresh master codes....... and runs into this error............ Isn't flutter release a stable version? Why still so big API changes

**Comment by zoechi** [comment]
`stable` is stable, `master` is not ;-)

**Comment by cbracken** [comment]
@jinfagang specifically, to switch to `stable` type `flutter channel stable`.

**Comment by mklim** [comment]
It looks like flutter/flutter#23759 was reverted by 630b8d843a1bd2a2df7136edaf64e123456bf8ee yesterday around 9AM -0800. @jinfagang based on the timing it sounds like you're probably still seeing the error even with the revert, but could you confirm that you're seeing problems even including 630b8d843a1bd2a2df7136edaf64e123456bf8ee? @alexleeelkins it would be great if you could double check and see if you're still seeing this with the latest commits too.

/cc @liyuqian

**Comment by liyuqian** [comment]
@mklim : I'm sorry for the confusion but https://github.com/flutter/flutter/commit/630b8d843a1bd2a2df7136edaf64e123456bf8ee is actually not merged into our tree. It was automatically generated by a Github revert PR when I try to revert something else. I then closed that revert PR (https://github.com/flutter/flutter/pull/25562) and created my own manual revert PR (https://github.com/flutter/flutter/pull/25563). In short, https://github.com/flutter/flutter/pull/23759 is not reverted.

**Comment by mklim** [comment]
Oh, my mistake. I should have actually double checked the diff too. Thanks @liyuqian.

@alexleeelkins and @jinfagang disregard what I said in https://github.com/flutter/flutter/issues/22467#issuecomment-449096452 for now, the likely issue hasn't been reverted yet.

**Comment by xster** [comment]
@alexleeelkins @jinfagang can either of you describe a bit more how we can reproduce this issue? 

Are your flutter_gallery and the flutter framework itself both from the same git repo?

**Comment by xster** [comment]
Removing the today tag since it's not reproducible in common usages.

**Comment by jeffersonatsafe** [comment]
I ran into this issue as well, but I've since resolved it (and have learned something new!)

This happened (in my case) because I cloned flutter (which defaulted to master branch), while my flutter installation was still in beta channel (I was unaware of flutter's four channels)

after learning about the existence of channels, this was fixed by a simple `git checkout origin/beta` in my local flutter repo. 
(or, if you're more adventurous, switching your flutter channel to master will also fix this)
basically, like @mklim said, you'd want your flutter branch to match your flutter channel

**Comment by zoechi** [comment]
Duplicate of #25220
